### PR TITLE
[feat] Make smart contract upgradeable + add editUser function

### PR DESCRIPTION
**Pull request by jsanchez034**
This change adds the editUser function to the smart contract to allow for user's to edit their profile. The smart contract was refactored to be upgradeable as well using OpenZeppelin Upgrade Plugin and the proxy pattern..
https://docs.openzeppelin.com/upgrades-plugins/1.x/

### CODE REVIEW

File: contracts/package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "compile": "npx hardhat compile",
     "deploy": "npx hardhat run scripts/deploy.js --network mumbai",
+    "upgrade": "pnpm compile && npx hardhat run scripts/upgrade.js --network mumbai",
```
Comment (eduairet) [inline contracts/package.json:9]: What a cool thing!

File: contracts/test/FontProject.proxy.js [hunk 2]
```diff
@@ -86,24 +90,55 @@ describe('Font Project Contract', () => {
...
-    await expect(fontProjectContract.connect(bob).createUser(
+    await expect(fontProjectContractInstance.connect(bob).createUser(
       lensHandle,
       profileInfoCID,
       createdAt
     )).to.be.revertedWith(
       'USER IS ALREADY REGISTERED'
     );
   });
+  it('should allow for editing a user', async () => {
```
Comment (eduairet) [inline contracts/test/FontProject.proxy.js:109]: Never saw this it() function before, is it something related with testing?

File: frontend/components/UI/NavBar.js [hunk 3]
```diff
@@ -41,21 +41,6 @@ export default function NavBar(props) {
...
-                variables: {
-                    owner: lensaddress,
-                },
-            });
-
-            const lensHandle = profileData?.profiles?.items[0]?.handle;
-
-            if (lensHandle) {
-                await createIPFontsUser({ lensAddress: lensaddress, lensHandle });
-            }
```
Comment (eduairet) [inline frontend/components/UI/NavBar.js:58]: Will this logic be moved to the same place where the user metadata upload happens?

**Review by eduairet — APPROVED** [review]
Thanks for this JP added comments on some lines of code, perhaps is good timing to merge @miglconts latest commit and this one, what do you all think?

**Review by hyperalchemy — APPROVED** [review]

File: contracts/package.json (hunk 1 above)
Comment (jsanchez034) [reply contracts/package.json:9]: Yeah this ability to upgrade smart contracts using a proxy contract is pretty awesome 😄 Reminds me of an ngnix reverse proxy in web2

File: frontend/components/UI/NavBar.js (hunk 3 above)
Comment (jsanchez034) [reply frontend/components/UI/NavBar.js:58]: Yeah exactly so we would call `createIPFontsUser` when a user updates their lens profile for the first time through our dapp if they have a lens handle or the user creates a new profile when the they don't have a lens handle.

File: contracts/package.json (hunk 1 above)
Comment (eduairet) [reply contracts/package.json:9]: Interesting! I'll research more about it 😃

File: contracts/test/FontProject.proxy.js (hunk 2 above)
Comment (jsanchez034) [reply contracts/test/FontProject.proxy.js:109]: Yeah so the it() function comes from [the Mocha testing framework](https://mochajs.org/) which is what hardhat uses under the hood to run unit tests. The it() function defines a test case and the describe() function groups a set of test cases. describe() is used to organize test cases so the test output is easier to understand...

https://mochajs.org/#bdd

File: contracts/package.json (hunk 1 above)
Comment (jsanchez034) [reply contracts/package.json:9]: Cool yeah this is a good starting point 😄 ...
https://docs.openzeppelin.com/upgrades-plugins/1.x/

File: contracts/test/FontProject.proxy.js (hunk 2 above)
Comment (eduairet) [reply contracts/test/FontProject.proxy.js:109]: Lovely!

**Pull request merged**